### ISSUE TITLE
[iTerm] Open new window command sets NODE_ENV to production

### ISSUE THREAD

**Issue by QuentinGaladrim** [raycast/extensions#10078]
### Extension

https://www.raycast.com/ron-myers/iterm

### Description

When using `New iTerm Window`, it sets node environment variable `NODE_ENV` to production when a launch from the app doesn't.
Using iTerm extension command :
![image](https://github.com/raycast/extensions/assets/112630508/b0a19a41-97fa-4b45-8019-1cd5a48c345b)

Using iTerm :
![image](https://github.com/raycast/extensions/assets/112630508/537f3aa1-6f56-4ad7-8ab3-0b8dce5840b9)

It induces error in Node project using `npm install` or `pnpm install` as these don't install dev dependencies.

### Steps To Reproduce

1. Launch New iTerm Window Command with iTerm being closed
2. Type `echo $NODE_ENV`

### Current Behaviour

Prints 'production'

### Expected Behaviour

Prints nothing

**Comment by QuentinGaladrim** [comment]
Hi !
Yes, this is true and this is the behavior I'm looking for too. I am not setting `NODE_ENV` anywhere and don't want it to be set. But when I'm launching iTerm using extension command `new iTerm window`, it does set the `NODE_ENV` variable to production.
Thus, there is nothing in my `~/.zshrc` that sets the variable.

**Comment by flops** [comment]
Wow!
Very sorry, firstly i thought the problem is actually opposite.
I've reproduced this bug.
First thing first, the problem is actually in `run-applescript` library and that raycast is leaking it's own env variables into process.
As we can see here [run-applescript/index.js](https://github.com/sindresorhus/run-applescript/blob/9db60e8a8fa7db46534c3c8a05c0f58135280ebb/index.js#L14C25-L14C38) it's using node `execFile` now and in version 6.1.0 it were using `execa` library.
But the issue is the same it's executes script with `process.env` and when raycast is running extension is passing `NODE_ENV` env variable to it for own reasons like show some additional developer actions in developing mode.
Long story short this kind of bug is actually affecting every extension and if there any sub process running that is not suited for this kind of variables it will be the same.

**Comment by flops** [comment]
WIP
